lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,23 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion;
+    const prerelease = semver.prerelease(currentValue);
+    let snapshot = false;
+    if (!prerelease || (snapshot = isSnapshot(prerelease))) {
```

Comment:
```suggestion
    let snapshot = prerelease ? isSnapshot(prerelease) : false;
    if (!prerelease || snapshot) {
```

shouldn't we go to else if `prerelease` is undefined | null? as we've a non prerelease semver?

Reply:
> shouldn't we go to else if `prerelease` is undefined | null? as we've a non prerelease semver?

A normal maven user would expect that bumping to a prerelease sets it to a SNAPSHOT version (https://github.com/renovatebot/renovate/pull/14763#discussion_r1000210402).
The else part is only for not normal (release or snapshot) maven versions, as it is hard to say what is expected in this case.